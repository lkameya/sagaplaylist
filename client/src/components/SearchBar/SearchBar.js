import React, { useState } from 'react';
import styled from 'styled-components';
import SearchButton from '../Icons/SearchButton/SearchButton';

const SearchBarWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 18rem;

  @media (max-width: 37.5em) {
    margin-top: 10rem;
  }
`;

const Input = styled.input`
  color: #191414;
  background-color: #f4f2f2;
  font-size: 1.6rem;
  border: none;
  padding: 1rem 4rem;
  border-radius: 10rem;
  width: 35%;
  transition: all .2s;
  margin-right: -5rem;

  &:focus {
    outline: none;
    width: 45%;
    background-color: #f0eeee;
  }

  &::-web-kit-input-placeholder {
    font-weight: 100;
    color: #ccc;
  }
`;

const SearchBar = props => {
  const [term, setTerm] = useState('');

  const onInputChange = (term) => {
    setTerm(term);
    props.handleSearch(term);
  }

  return (
    <SearchBarWrapper>
      <Input
        type="text"
        value={term}
        placeholder="Search an artist or song ..."
        onChange={event => onInputChange(event.target.value)}
      />
      <SearchButton />
    </SearchBarWrapper>
  )
}

export default SearchBar;

