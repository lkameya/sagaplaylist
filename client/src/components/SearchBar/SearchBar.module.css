.search {
  flex: 0 0 40%;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #191414;
}

.input {
  color: #191414;
  background-color: #f4f2f2;
  font-size: 1.6rem;
  border: none;
  padding: 1rem 4rem;
  border-radius: 10rem;
  width: 35%;
  transition: all .2s;
  margin-right: -5rem;
}

.input:focus {
  outline: none;
  width: 45%;
  background-color: #f0eeee;
}

.input::-web-kit-input-placeholder {
  font-weight: 100;
  color: #ccc;
}

.input:focus + .button {
  background-color: #f0eeee;
}

.searchBar {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 18rem;
}

/* @media only screen and (max-width: 37.5em) {  
  .searchBar {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-top: 10rem;
  }
};     */
