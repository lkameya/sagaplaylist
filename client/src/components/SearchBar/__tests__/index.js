import React from 'react';
import SearchBar from '../SearchBar';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Search Bar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SearchBar>
      </SearchBar>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});