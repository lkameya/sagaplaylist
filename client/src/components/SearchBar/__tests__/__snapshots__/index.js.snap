// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar renders correctly 1`] = `
.c2 {
  margin-left: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 5rem;
}

.c1 {
  color: #191414;
  background-color: #f4f2f2;
  font-size: 1.6rem;
  border: none;
  padding: 1rem 4rem;
  border-radius: 10rem;
  width: 35%;
  -webkit-transition: all .2s;
  transition: all .2s;
  margin-right: -5rem;
}

.c1:focus {
  outline: none;
  width: 45%;
  background-color: #f0eeee;
}

.c1::-web-kit-input-placeholder {
  font-weight: 100;
  color: #ccc;
}

@media (max-width:37.5em) {
  .c0 {
    margin-top: 10rem;
  }
}

<div
  className="c0"
>
  <input
    className="c1"
    onChange={[Function]}
    placeholder="Search an artist or song ..."
    type="text"
    value=""
  />
  <div
    className="c2"
  >
    <svg>
      search.svg
    </svg>
  </div>
</div>
`;
